http/sys_raft.go

```diff
@@ -21,7 +23,30 @@ func handleSysRaftBootstrap(core *vault.Core) http.Handler {
 				respondError(w, http.StatusBadRequest, errors.New("node must be unsealed to bootstrap"))
 			}
 
-			if err := core.RaftBootstrap(context.Background(), false); err != nil {
+			// Parse the request
+			var req map[string]interface{}
+			if _, err := parseJSONRequest(core.PerfStandby(), r, w, &req); err != nil {
+				respondError(w, http.StatusBadRequest, err)
+				return
+			}
+
+			resetTLSKeyring := req["reset_tls_keyring"]
+			// It defaults to false if not set.
+			if resetTLSKeyring == nil {
+				resetTLSKeyring = false
+			}
+			// Convert it to bool if needed because vault client sends it as string even if its a boolean but vault API sends it as bool.
+			if reflect.TypeOf(resetTLSKeyring) == reflect.TypeOf("string") {
+				var err error
+				resetTLSKeyring, err = strconv.ParseBool(resetTLSKeyring.(string))
```

Comment:
Instead of this new parameter, I think we should dynamically determine in `raftCreateTLSKeyring` whether or not we're using raft storage. This avoids the need for a new option, and all of the added code in this area